### ISSUE TITLE
Cursor does not jump to the end of the eshell prompt when entering hybrid mode from normal mode.

### ISSUE THREAD

**Issue by r-darwish** [syl20bnr/spacemacs#6331]
#### Description :octocat:
Cursor does not jump to the end of the eshell prompt when entering hybrid mode from normal mode.

#### Reproduction guide :beetle:
- Start Emacs (Use hybrid mode instead of vim mode)
- Start Eshell
- Escape to normal mode
- Move the cursor away from the end of the prompt
- Enter hybrid mode by pressing `i`

*Observed behaviour:* :eyes: :broken_heart:
The cursor does not jump to the end of the prompt.

*Expected behaviour:* :heart: :smile:
The cursor should jump to the end of the prompt as happens when entering insert mode.

#### System Info :computer:
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: develop (rev. 8368be3)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:
```elisp
(osx darwish ansible
     (auto-completion :variables auto-completion-private-snippets-directory dcp-snippet-dir auto-completion-complete-with-key-sequence-delay 0.0 auto-completion-enable-snippets-in-popup t)
     emacs-lisp
     (go :variables go-tab-width 4)
     restclient dockerfile search-engine ivy python javascript html git org github markdown semantic rust c-c   systemd
     (shell :variables shell-default-shell 'eshell)
     colors
     (syntax-checking :variables syntax-checking-enable-tooltips nil)
     (version-control :variables version-control-diff-tool 'diff-hl))
```

**Comment by syl20bnr** [comment]
should be fixed in develop

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
